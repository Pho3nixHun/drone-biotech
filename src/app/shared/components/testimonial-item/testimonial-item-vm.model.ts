export interface TestimonialItemVM {
    messageKey: string;
    name: string;
    roleAndCompanyKey: string;
}
