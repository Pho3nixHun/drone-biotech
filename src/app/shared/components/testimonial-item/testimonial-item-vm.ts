export interface TestimonialItemVM {
  message: string;
  name: string;
  location: string;
}
