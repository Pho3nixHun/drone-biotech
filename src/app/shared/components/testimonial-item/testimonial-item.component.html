@if (vm(); as vm) {
  <div class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer">
    <p class="text-lg italic text-gray-700">"{{ vm.message }}"</p>
    <p class="font-semibold text-gray-900 mt-4">{{ vm.name }}</p>
    <p class="text-gray-500">{{ vm.location }}</p>
  </div>
}
