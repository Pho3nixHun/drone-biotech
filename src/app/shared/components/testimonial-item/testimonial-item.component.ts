import { Component, input } from '@angular/core';
import { TestimonialItemVM } from './testimonial-item-vm.model';
import { TranslocoModule } from '@jsverse/transloco';

/**
 * TestimonialItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual Testimonial Item.
 * - Displays the testimonial's message, name and location.
 * - Manages the styling and the layout of the Testimonial Item.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the testimonial.
 * - Does not manage or transform testimonial data.
 *
 * Purpose (optional):
 * - To render an individual Testimonial Item in a predefined layout.
 */

@Component({
    selector: 'app-testimonial-item',
    imports: [TranslocoModule],
    templateUrl: './testimonial-item.component.html'
})
export class TestimonialItemComponent {
    public vm = input.required<TestimonialItemVM>();
}
