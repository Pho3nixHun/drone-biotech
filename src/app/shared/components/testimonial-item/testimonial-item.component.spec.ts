import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TestimonialItemComponent } from './testimonial-item.component';
import { TestimonialItemVM } from './testimonial-item-vm.model';
import { getTranslocoModule } from 'transloco-testing.module';

const en = { message: 'Hello', roleAndCompany: 'World' };

describe('TestimonialItemComponent', () => {
  let component: TestimonialItemComponent;
  let fixture: ComponentFixture<TestimonialItemComponent>;
  let compiled: HTMLElement;
  const vm: TestimonialItemVM = {
    messageKey: en.message,
    name: 'Emily Johnson',
    roleAndCompanyKey: en.roleAndCompany,
  };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        TestimonialItemComponent,
        getTranslocoModule({
          langs: { en },
          translocoConfig: { availableLangs: ['en'], defaultLang: 'en' },
        }),
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(TestimonialItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  //Snapshot test
  it('should not render the template when there is not VM provided', () => {
    //Arrange

    //Act

    //Assert
    expect(compiled).toMatchSnapshot();
  });

  //Snapshot test
  it('should get the input data and visualize it', () => {
    //Arrange
    fixture.componentRef.setInput('vm', vm);

    //Act
    fixture.detectChanges();

    //Assert
    expect(compiled).toMatchSnapshot();
  });
});
