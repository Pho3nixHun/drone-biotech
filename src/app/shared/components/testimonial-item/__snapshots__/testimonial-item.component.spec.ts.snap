// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialItemComponent should get the input data and visualize it 1`] = `
<div
  id="root2"
>
  <div
    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
  >
    <p
      class="text-lg italic text-gray-700"
    >
      "Hello"
    </p>
    <p
      class="font-semibold text-gray-900 mt-4"
    >
      Emily Johnson
    </p>
    <p
      class="text-gray-500"
    >
      World
    </p>
  </div>
  
  
</div>
`;

exports[`TestimonialItemComponent should not render the template when there is not VM provided 1`] = `
<div
  id="root1"
>
  
</div>
`;
